source/Handlebars/Compiler/Lexer/Converter/BlockAccumulators/BlockAccumulatorContext.cs

```diff
@@ -77,6 +81,54 @@ private static bool IsPartialBlock (Expression item)
             }
         }
 
+        private static bool IsLooseClosingElement(Expression item, Expression parentItem, out string looseBlockName)
+        {
+            looseBlockName = null;
+
+            var itemBlockName = GetBlockName(item);
+
+            if (itemBlockName == null) return false;
+
+            var parentBlockName = GetBlockName(parentItem);
+
+            if (!itemBlockName.StartsWith("/")) return false;
+
+            if (parentBlockName == null || IsClosingBlockNotMatchParentBlock(itemBlockName, parentBlockName))
+            {
+                looseBlockName = itemBlockName;
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsClosingBlockNotMatchParentBlock(string itemBlockName, string parentBlockName)
+        {
+            if (itemBlockName == null) throw new ArgumentNullException(nameof(itemBlockName));
+            if (parentBlockName == null) throw new ArgumentNullException(nameof(parentBlockName));
```

Comment:
nit: Shouldn't it be `string.IsNullOrEmpty` check?

Reply:
I used this check to ensure that this function won't be used incorrectly in other places (just in case). It checks only null because `GetBlockName()` returns null as default (when a statement is nor PathExpression nor HelperExpression).